fixes of i18n on orders-feed.md (#423)
vtexdocs/dev-portal-content: docs/guides/Orders/orders-feed.md

In this sense, the feed is not a list of orders, but rather a list of events. For example, if the status of an order is changed to `Approve payment` and then to `Authorize shipping`, the feed will receive two events: one for each update, both related to the same order. You can configure the feed to filter the updates that will actually generate feed events, instead of having all updates in all orders generating events in the feed queue.

This guide explains how Feed and Hook work and how to configure them to build order integrations. Also, in the latter part of the article, we explain the differences between each and when to choose one over the other based on the specific needs of your operation.

## Order integration: Feed vs. List orders

Some stores use the List orders API request to check order status changes. However, this method only returns orders that have already been indexed, which may lead to some problems:

- If a problem in the system prevents indexing, you will not be able to consume the order list, and order updates may not be visible.

- Because the `list` method depends on indexing, it's slower and performs less than the feed.

> On the other hand, the feed has been specifically developed to track order updates. It runs before indexing and doesn't depend on it, making it the most reliable and fastest method to track order updates. [block:callout]

Reviewer: ```suggestion
On the other hand, the feed has been specifically developed to track order updates. It runs before indexing and doesn't depend on it, making it the most reliable and fastest method to track order updates.
```

> { "type": "warning", "body": "If you have an integration based on the List orders API request, you should migrate it to the feed. However, keep in mind that this means changing the integration flow. Read the [order integration guide](https://developers.vtex.com/docs/guides/erp-integration-set-up-order-integration) to learn how to implement this change.", "title": "Migrate your integration" } [/block]

Reviewer: ```suggestion
>⚠️ If you have an integration based on the List orders API request, you should migrate it to the feed. However, keep in mind that this means changing the integration flow. Read the [order integration guide](https://developers.vtex.com/docs/guides/erp-integration-set-up-order-integration) to learn how to implement this change.
```

## Access

Configuring and using Feed v3 and Hook is only allowed when authorization is granted in the Order Management module.

> The feed's application key must have a [role](https://help.vtex.com/en/tutorial/roles--7HKK5Uau2H6wxE1rH5oRbc#) with one of the appropriate resources: `Feed v3 and Hook Admin` or `Feed v3 and Hook view only`, depending on the intended use. [block:callout]

Reviewer: ```suggestion
The feed's application key must have a [role](https://help.vtex.com/en/tutorial/roles--7HKK5Uau2H6wxE1rH5oRbc#) with one of the appropriate resources: `Feed v3 and Hook Admin` or `Feed v3 and Hook view only`, depending on the intended use.
```

> {

Reviewer: ```suggestion

```

>   "type": "warning",

Reviewer: ```suggestion
 
```

>   "body": "Each appKey can configure or access only one feed. This means that different users sharing an appKey access the same feed. In this case, if a user commits an item to the queue, the item is removed from the feed and won't be available for any users sharing the same appKey. Therefore, we recommend configuring one feed per appKey per user, ensuring that each user has access to their own feed."

Reviewer: ```suggestion
>⚠️ Each appKey can configure or access only one feed. This means that different users sharing an appKey access the same feed. In this case, if a user commits an item to the queue, the item is removed from the feed and won't be available for any users sharing the same appKey. Therefore, we recommend configuring one feed per appKey per user, ensuring that each user has access to their own feed.
```

> }

Reviewer: ```suggestion
 
```

> [/block]

Reviewer: ```suggestion

```

> [block:callout]

Reviewer: ```suggestion

```

> { "type": "info", "body": "Before getting data from the order feed queue, you need to set it up. Otherwise, the API will return a `404 not found` error. Order updates will only show in the feed queue only after it's configured. You can learn more about configuring the feed below." } [/block]

Reviewer: ```suggestion
>ℹ️ Before getting data from the order feed queue, you need to set it up. Otherwise, the API will return a `404 not found` error. Order updates will only show in the feed queue only after it's configured. You can learn more about configuring the feed below.
```

```

> [block:callout]

Reviewer: ```suggestion
 
```

> { "type": "info", "body": "The `FromWorkflow` filter may be limited for some of the integration needs of your store. If you want more filter options, consider using the `FromOrders` filter." } [/block]

Reviewer: ```suggestion
>ℹ️ The `FromWorkflow` filter may be limited for some of the integration needs of your store. If you want more filter options, consider using the `FromOrders` filter.
```

```

> [block:callout]

Reviewer: ```suggestion

```

> { "type": "warning", "body": "The two filter types are mutually exclusive. If you pass `FromWorkflow` in the `type` field, you should not include the `expression` field nor `disableSingleFire` in the body. The reverse is also true. If you pass `FromOrders` `type`, you should not include the `status` field. If you combine them, you will get a `409 conflict` status response." } [/block]

Reviewer: ```suggestion
>⚠️ The two filter types are mutually exclusive. If you pass `FromWorkflow` in the `type` field, you should not include the `expression` field nor `disableSingleFire` in the body. The reverse is also true. If you pass `FromOrders` `type`, you should not include the `status` field. If you combine them, you will get a `409 conflict` status response.
```

> [block:callout]

Reviewer: ```suggestion
 
```

> { "type": "info", "body": "If the `filter` is not configured, the `FromWorkflow` type is used, and all status changes will appear in the feed." } [/block]

Reviewer: ```suggestion
>ℹ️ If the `filter` is not configured, the `FromWorkflow` type is used, and all status changes will appear in the feed.
```

> [block:callout]

Reviewer: ```suggestion

```

> {

Reviewer: ```suggestion
 
```

>   "type": "info",

Reviewer: ```suggestion
  
```

>   "body": "When a new feed is configured, its queue contains whatever orders are changed right after the setup is complete. If the feed is reconfigured, events from the former queue will remain in the feed until they are committed or until the retention period expires."

Reviewer: ```suggestion
 >ℹ️ When a new feed is configured, its queue contains whatever orders are changed right after the setup is complete. If the feed is reconfigured, events from the former queue will remain in the feed until they are committed or until the retention period expires.
```

> }

Reviewer: ```suggestion
 
```

> [/block]

Reviewer: ```suggestion
 
```

> ❗ If the feed doesn't receive any new events in its queue during the time set in `messageRetentionPeriodInSeconds`, your configuration will be removed, and you will have to reconfigure it with the [Feed configuration API call](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/orders/feed/config) to continue using the feed. Therefore, it's important to be mindful of the filter configuration you are using. You can check it any time using the [Get feed configuration](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/orders/feed/config) endpoint.

## Feed readout

> Every system that depends on order updates should consume the feed to be able to take the necessary actions based on that information. The most common behavior is a store system or an integration reading every event in the feed and, based on its status, making a decision for each one. [block:callout]

Reviewer: ```suggestion
Every system that depends on order updates should consume the feed to be able to take the necessary actions based on that information. The most common behavior is a store system or an integration reading every event in the feed and, based on its status, making a decision for each one. 
```

> { "type": "warning", "body": "When filtering statuses, be aware that all possible order statuses must be dealt with during integration to avoid errors. Particular attention should be paid to `Status Null`, which may be unidentified and end up being mapped as another status, which can potentially lead to errors." } [/block]

Reviewer: ```suggestion
>⚠️ When filtering statuses, be aware that all possible order statuses must be dealt with during integration to avoid errors. Particular attention should be paid to `Status Null`, which may be unidentified and end up being mapped as another status, which can potentially lead to errors.
```

### Example

The order feed can be useful in many ways. For instance, say you want to develop an integration between your ERP and VTEX. This integration could have the following behavior:

1. It retrieves ten events from the feed.
2. For each one of the ten events, it evaluates the status the order was changed to.
3. If the order changed to the status `Ready for handling`, the integration gets the complete order data, records it in the ERP, and updates the status on VTEX to `Handling shipping` (which indicates the start of handling).
4. Then, it commits the events to the Feed (removing them from the queue).
5. After reading and removing these ten events from the list, the integration moves to the following ten events in the feed and repeats the process.

![feedFlow-min](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/images/orders-feed-0.png)

Check the [back-office order integration guide](https://developers.vtex.com/docs/guides/erp-integration-set-up-order-integration) to learn more about how you can integrate the VTEX order feed with an ERP.

## Hook

> Hook is a counterpart to Feed. It allows integrations to consume order update data differently. Instead of receiving events to form a queue that can be retrieved, a hook automatically sends the items to a URL provided by the user in the hook configuration. [block:callout]
> { "type": "warning", "body": "Since Hook is a counterpart, [access](#access) follows the same principles described for Feed above. This means each appKey can configure or access only one hook. Different users that sharing the same appKey access the same hook. We recommend configuring one hook per appKey per user, ensuring that each user has access to their own hook." } [/block]

Reviewer: ```suggestion
Hook is a counterpart to Feed. It allows integrations to consume order update data differently. Instead of receiving events to form a queue that can be retrieved, a hook automatically sends the items to a URL provided by the user in the hook configuration. 

>⚠️ Since Hook is a counterpart, [access](#access) follows the same principles described for Feed above. This means each appKey can configure or access only one hook. Different users that sharing the same appKey access the same hook. We recommend configuring one hook per appKey per user, ensuring that each user has access to their own hook.
```

```

> [block:callout]

Reviewer: ```suggestion
 
```

> { "type": "warning", "body": "The given endpoint should return status 200 for the above-mentioned request. Otherwise, the Hook API will return status `400 Bad Request`, and you won't be able to save the configuration." } [/block]

Reviewer: ```suggestion
>⚠️ The given endpoint should return status 200 for the above-mentioned request. Otherwise, the Hook API will return status `400 Bad Request`, and you won't be able to save the configuration.
```

> [block:callout]

Reviewer: ```suggestion
 
```

> {

Reviewer: ```suggestion
 
```

>   "type": "warning",

Reviewer: ```suggestion
 
```

>   "body": "We recommend configuring the hook in the main account. This ensures more visibility to commit all events correctly and that the configured endpoint is more frequently notified, preventing the hook from being excluded due to inactivity."

Reviewer: ```suggestion
>⚠️ We recommend configuring the hook in the main account. This ensures more visibility to commit all events correctly and that the configured endpoint is more frequently notified, preventing the hook from being excluded due to inactivity.
```

> }

Reviewer: ```suggestion
 
```

> [/block]

Reviewer: ```suggestion
 
```

### Hook notifications

If configured, the Hook notifies the integration endpoint whenever an order update is made and meets the conditions specified in the `filter`.

If a new event is not correctly notified to the endpoint, the interval for future retries is recalculated based on an internal geometric progression algorithm.
> ❗ If the hook has no notifications for three days, your configuration will be removed, and you will have to reconfigure it with the [Hook configuration API call](https://developers.vtex.com/docs/api-reference/orders-api#post-/api/orders/hook/config) to continue using it. Therefore, it's important to be mindful of your filter configuration. You can check it any time using the [Get hook configuration](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/orders/hook/config) endpoint.

> [block:callout]

Reviewer: ```suggestion
 
```

> {

Reviewer: ```suggestion
 
```

>   "type": "warning",
>   "body": "When notified, the configured endpoint must always respond with HTTP status 200 within 5000 ms."
> }

Reviewer: ```suggestion
>⚠️ When notified, the configured endpoint must always respond with HTTP status 200 within 5000 ms.
```

> [/block] Below is an example of a hook notification request body made to the integration endpoint.

Reviewer: ```suggestion
Below is an example of a hook notification request body made to the integration endpoint.
```

}
```

>   "body": "When using a hook, it's important to be aware that it's a reactive feature. This means your middleware or ERP system must be ready to deal with whatever volume of data the hook sends. Large peaks in sales – due to Black Friday, for example – tend to increase hook notifications. If the implementation is not prepared for this peak, it may cause problems in the integration, compromising the store’s ability to handle orders and receive further notifications. Learn more about how to deal with this issue in the next section."

Reviewer: ```suggestion

```

> }

Reviewer: ```suggestion

```

> [/block]

Reviewer: ```suggestion

```

> [block:callout]
> {
>   "type": "warning",

Reviewer: ```suggestion

```

>   "body": "When using a hook, it's important to be aware that it's a reactive feature. This means your middleware or ERP system must be ready to deal with whatever volume of data the hook sends. Large peaks in sales – due to Black Friday, for example – tend to increase hook notifications. If the implementation is not prepared for this peak, it may cause problems in the integration, compromising the store’s ability to handle orders and receive further notifications. Learn more about how to deal with this issue in the next section."
> }
> [/block]

Reviewer: ```suggestion
>⚠️ When using a hook, it's important to be aware that it's a reactive feature. This means your middleware or ERP system must be ready to deal with whatever volume of data the hook sends. Large peaks in sales – due to Black Friday, for example – tend to increase hook notifications. If the implementation is not prepared for this peak, it may cause problems in the integration, compromising the store’s ability to handle orders and receive further notifications. Learn more about how to deal with this issue in the next section.
```
